include/magic_enum.hpp

```diff
@@ -1225,7 +1243,7 @@ template <detail::enum_subtype S, typename E>
 // Returns name from enum-flags value.
 // If enum-flags value does not have name or value out of range, returns empty string.
 template <typename E>
-[[nodiscard]] auto enum_flags_name(E value, char sep = '|') -> detail::enable_if_t<E, string> {
+[[nodiscard]] auto enum_flags_name(E value, char_type sep = '|') -> detail::enable_if_t<E, string> {
```

Comment:
static_cast<char_type>('-') 

Check this character too at static_assert check at line 141/142